core/frontend/src/tools/ToolAdmin.ts

```diff
@@ -1230,6 +1275,8 @@ export class ToolAdmin {
     current.toEvent(ev, true);
     current.updateDownPoint(ev);
 
+    await this.onMotionSnap(ev);
```

Comment:
This looks wrong, seems like onMotionSnap should be called before toEvent so that the BeButtonEvent accounts for the correct snap location...it also seems like this doesn't ensure that _lastHandledMotionTime is undefined to ensure that onMotionSnapOrSkip doesn't early return?

Have these changes been tested with touch input...I can't test those now as I don't have a touch screen...

Reply:
Isn't ``toEvent`` initializing ``ev``? If so, we cannot call ``onMotionSnap`` before. In ``onMotion``, it seeems that toEvent is called once to with useSnap false before and once with useSnap true after the call to onMotionSnap. Should we do similarly here?
```
    current.toEvent(ev, false);
    current.updateDownPoint(ev);
    this._lastHandledMotionTime  = undefined;
    await this.onMotionSnap(ev);
    current.toEvent(ev, true);
```


And indeed, there is no reason to not reset _lastHandledMotionTime anymore. I will fix this.

I have not been able to test with touch screen either, other than with built-in desktop browser tools.

Reply:
Yes, toEvent initializes ev, so it would have to be called with false, then call onMotionSnap, then call again with true (basically what happens in onMotion).